### PR TITLE
Added functionality

### PR DESCRIPTION
**Pull request by christophedemey**
- Ability to change local table data through the data property.
- Ability to fresh the table from local table data.
In my scenario i wanted when a user deletes a row the row would be immediately gone and not wait for the server request to finish.

- Ability to fresh the table from server data.
In my scenario once the row was effectively deleted on the server side i wanted to do a server side refresh.

- dataLoading event with boolean indicating status, true=loading/false=finished.
I needed to show a loader while server side actions are being performed such as paging, initial data load, sorting.

### CODE REVIEW

**Comment by christophedemey** [comment]
Hi @relair 

Any feedback ? :)

**Comment by pnguyen535** [comment]
I need this functionality as well. Thanks

**Comment by relair** [comment]
I'll try to look at that soon. I've been busy lately.

File: projects/odata-data-source/src/lib/odata-data-source.ts [hunk 1]
```diff
@@ -22,14 +25,33 @@ export class ODataDataSource extends DataSource<any> {
   constructor(
     private readonly httpClient: HttpClient,
     private readonly resourcePath: string) {
-      super();      
+    super();
+
+    this.data = new Array<any>();
```
Comment (relair) [inline projects/odata-data-source/src/lib/odata-data-source.ts]: I don't really like this way of doing it, it creates additional source of data that doesn't need to consistent with data from dataSubject. Maybe instead we can just have an update function that can update the data and not store extra data inside, especially not public.

**Comment by relair** [comment]
I've released version 1.2.0, it should give you the ability to do all those things.

**Pull request closed without merging**